cmscontrib/loaders/lio.py

```
@@ -0,0 +1,322 @@
+        for line in content:
+            vars = line.replace("-", " ").split()
+            a = int(vars[0])
+            b = int(vars[1])
+            points = int(vars[2])
+            for group in range(a, b+1):
+                if group in points_per_group:
+                    logger.critical("Duplicated groups in point file")
+                    raise ValueError
+                points_per_group[group] = points
+        for group in range(len(points_per_group)):
+            if group not in points_per_group:
+                logger.critical("Missing group from point file")
+                raise ValueError
+        return points_per_group
+
+
+    def get_task(self, get_statement):
+        args = {
+            'name': self.conf['name'],
+            'title': self.conf['title'],
+        }
+        name = args['name']
+
+        logger.info(f"Loading parameters for task {name}")
+
+        if get_statement:
+            args['statements'] = {}
+            for statement in self.conf.get('statements', []):
+                path, lang = statement
+                logger.info(f"Loading statement: {statement}")
+                digest = self.file_cacher.put_file_from_path(
+                    os.path.join(self.task_dir, path),
+                    f"Statement for task {name} (lang: {lang})",
+                )
+                args['statements'][lang] = Statement(lang, digest)
+            if args['statements']:
+                args['primary_statements'] = self.conf.get('primary_statements', ["lv"])
+
+        args['submission_format'] = [f"{name}.%l"]
+
+        set_if_present(self.conf, args, 'score_precision')
+
+        score_mode = self.conf.get('score_mode', SCORE_MODE_MAX_TOKENED_LAST)
+        if score_mode in [SCORE_MODE_MAX, SCORE_MODE_MAX_SUBTASK, SCORE_MODE_MAX_TOKENED_LAST]:
+            args['score_mode'] = score_mode
+        else:
+            logger.critical("Unknown score mode provided")
+            raise ValueError
```

Some places in this method raise an exception and some others return `None` on error. Which is the expected way?

---

I confused this with `get_task`, `get_contest` and similar functions that accepts returned None, will fix this.

---

italy loader were ignoring errors in this way. I used the same behaviour in the most part. Will replace with correctly raised exceptions where necessary.